### PR TITLE
Start password strength checking for update_password function

### PR DESCRIPTION
**Pull request by 4855062503**
Not yet finished adding strength-checking but works so far.

Will continue to update this PR.

### CODE REVIEW

**Comment by djpnewton** [comment]
You should use the same password validation code as flask security (so the behaviour is the same in app and on the website)

https://flask-security-too.readthedocs.io/en/stable/patterns.html?highlight=password#password-validation-and-complexity


You can set the settings in `flask_config.py` and use the functions: `flask_security.password_length_validator() flask_security.password_complexity_validator() flask_security.password_breached_validator()`

**Comment by 4855062503** [comment]
Do we want to use this function?

https://flask-security-too.readthedocs.io/en/stable/api.html#flask_security.password_breached_validator

Will require an API key

**Comment by djpnewton** [comment]
> Do we want to use this function?
> 
> https://flask-security-too.readthedocs.io/en/stable/api.html#flask_security.password_breached_validator
> 
> Will require an API key

yep, it will only do something if the setting is conifigured

File: api_endpoint.py [hunk 1]
```diff
@@ -344,6 +344,8 @@ def user_update_password():
     if not verified_password:
         return bad_request(web_utils.INCORRECT_PASSWORD)
     ### set the new_password:
+    if (flask_security.password_length_validator(new_password) is not None) or (flask_security.password_complexity_validator(new_password, True) is not None) or (flask_security.password_breached_validator(new_password) is not None):
```
Comment (djpnewton) [inline api_endpoint.py:347]:
```suggestion
    if flask_security.password_length_validator(new_password) or flask_security.password_complexity_validator(new_password, True) or flask_security.password_breached_validator(new_password):
```

**Review by djpnewton — CHANGES_REQUESTED** [review]

**Comment by 4855062503** [comment]
Updated PR.

File: api_endpoint.py [hunk 2]
```diff
@@ -344,6 +344,8 @@ def user_update_password():
     if not verified_password:
         return bad_request(web_utils.INCORRECT_PASSWORD)
     ### set the new_password:
+    if flask_security.password_length_validator(new_password) is not None or flask_security.password_complexity_validator(new_password, True) is not None or flask_security.password_breached_validator(new_password) is not None:
```
Comment (djpnewton) [inline api_endpoint.py:347]:
```suggestion
    if flask_security.password_length_validator(new_password) or flask_security.password_complexity_validator(new_password, True) or flask_security.password_breached_validator(new_password):
```

**Review by djpnewton — CHANGES_REQUESTED** [review]
you didnt make all the changes

**Comment by 4855062503** [comment]
Updated PR.

**Pull request merged**